package/butterfly/lib/dev_server.dart

```diff
@@ -123,29 +120,36 @@ class ButterflyDevServer {
     request.response.write(JSON.encode(result));
   }
 
-  Future<Null> _serveDevServerClientFile(HttpRequest request) async {
+  /// Mimics pub serve's behavior of service package-relative static files.
+  Future<Null> _servePackageAsset(HttpRequest request) async {
     final packagesFile = new File('.packages');
 
     if (!await packagesFile.exists()) {
       throw new StateError('.packages file not found. Have you run pub get?');
     }
 
+    final pathFragments = pathlib.split(request.uri.path);
+    assert(pathFragments[0] == '/');
+    assert(pathFragments[1] == 'packages');
+    final packageName = pathFragments[2];
+    final pathWithinPackage = pathFragments.skip(3).join(pathlib.separator);
+
     List<String> packageInfoParts = await packagesFile
         .openRead()
         .transform(const Utf8Decoder())
         .transform(const LineSplitter())
         .map((line) => line.split(':'))
-        .firstWhere((parts) => parts[0] == 'butterfly',
+        .firstWhere((parts) => parts[0] == packageName,
             defaultValue: () => null);
 
     if (packageInfoParts == null) {
       throw new StateError(
-          '"butterfly" package not found in .packages file. Please check your pubspec.yaml and run pub get again.');
+          '"$packageName" package not found in .packages file. Please check your pubspec.yaml and run pub get again.');
     }
 
-    final file = new File('${packageInfoParts.last}/${_devServerClientFile}');
+    final file = new File('${packageInfoParts.last}/${pathWithinPackage}');
```

Comment:
Here and elsewhere (at least for server code), replace with `Platform.pathSeparator`

Reply:
Done.